internal/probe/probe.go

```diff
@@ -294,14 +305,59 @@ func (p *Probe) attach() error {
 	}
 
 	if len(p.openFilesToTrack) > 0 {
-		// attach to openat syscall
-		l, err = link.Tracepoint("syscalls", "sys_enter_openat", p.c.Programs[fileOpenProgramName], nil)
+		err = p.attachOpenPrograms()
 		if err != nil {
-			return fmt.Errorf("can't attach probe sys_enter_openat: %w", err)
+			return fmt.Errorf("failed to attach file open tracking programs: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (p *Probe) attachOpenPrograms() error {
```

Comment:
Set the programs as map of enter/exit and iterate the map instead of copypasta ;)